.github/workflows/announce-release.yml

```
@@ -0,0 +1,35 @@
+name: Announce Release on Mastodon
+
+on:
+  pull_request:
+    types:
+      - closed
+    branches:
+      - main
```

I think this is the better way to target merges into the main branch, and to restrict the workflow to only run when there have been changes to the `CHANGELOG.md`:
```suggestion
  push:
    branches:
      - main
    path:
      - 'CHANGELOG.md'
```
Here's some background on this:
https://stackoverflow.com/a/61565445

And I would still leave in the check `chore(main): release` check for the actual job, to avoid toots when someone manually edits the `CHANGELOG.md`.